### ISSUE TITLE
BorrowMutError panic when calling Deno.utime and Deno.open together

### ISSUE THREAD

**Issue by mk12** [denoland/deno#8982]
I reliably get a Rust panic from the following code.

### Repro

```typescript
// bug.ts
let now = new Date();
Deno.utime("foo", now, now);
Deno.open("bar");
```

```
$ deno run --unstable --allow-write bug.ts
thread 'main' panicked at 'already borrowed: BorrowMutError', cli/ops/fs.rs:203:53
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

```
$ RUST_BACKTRACE=1 deno run --unstable --allow-write repro.ts
thread 'main' panicked at 'already borrowed: BorrowMutError', cli/ops/fs.rs:203:53
stack backtrace:
   0: _rust_begin_unwind
   1: core::panicking::panic_fmt
   2: core::result::unwrap_failed
   3: core::cell::RefCell<T>::borrow_mut
   4: <futures_util::future::future::Map<Fut,F> as core::future::future::Future>::poll
   5: <futures_util::future::future::Inspect<Fut,F> as core::future::future::Future>::poll
   6: <futures_util::future::future::Map<Fut,F> as core::future::future::Future>::poll
   7: <futures_util::stream::futures_unordered::FuturesUnordered<Fut> as futures_core::stream::Stream>::poll_next
   8: deno_core::runtime::JsRuntime::poll_event_loop
   9: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
  10: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
  11: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
  12: tokio::runtime::Runtime::block_on
  13: deno::main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

```
$ RUST_BACKTRACE=full deno run --unstable --allow-write repro.ts
thread 'main' panicked at 'already borrowed: BorrowMutError', cli/ops/fs.rs:203:53
stack backtrace:
   0:        0x10df671f3 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h0afb3dc3ec8cd05f
   1:        0x10dbcff5e - core::fmt::write::h39441ef24fae20ea
   2:        0x10df66fb9 - std::io::Write::write_fmt::h2ffecc964e3c3ddd
   3:        0x10df868f5 - std::panicking::default_hook::{{closure}}::h1a491655bcf6394f
   4:        0x10df870ec - std::panicking::rust_panic_with_hook::h489020cfd35413ea
   5:        0x10df675ac - std::panicking::begin_panic_handler::{{closure}}::he498abc45ca35fbf
   6:        0x10df67577 - std::sys_common::backtrace::__rust_end_short_backtrace::h4a2a0fae6b0989d8
   7:        0x10df831a0 - _rust_begin_unwind
   8:        0x10eca80bf - core::panicking::panic_fmt::h0003130af3c08aa0
   9:        0x10eca8185 - core::result::unwrap_failed::hbfeca77107a6ab03
  10:        0x10d9a4b9d - core::cell::RefCell<T>::borrow_mut::h852b399ba3fa9d57
  11:        0x10da72418 - <futures_util::future::future::Map<Fut,F> as core::future::future::Future>::poll::h85d678d4ead040a9
  12:        0x10d8e485b - <futures_util::future::future::Inspect<Fut,F> as core::future::future::Future>::poll::h1bf754d869b6c7ed
  13:        0x10dbdf6ab - <futures_util::future::future::Map<Fut,F> as core::future::future::Future>::poll::h7b1b4f483bf8201a
  14:        0x10dbe1a89 - <futures_util::stream::futures_unordered::FuturesUnordered<Fut> as futures_core::stream::Stream>::poll_next::heb76d6b038ebee87
  15:        0x10dbefebb - deno_core::runtime::JsRuntime::poll_event_loop::habc970f7b5523ebb
  16:        0x10da5358d - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::hb33d7b30ee81c7bd
  17:        0x10da549af - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::hd2acc411b323eb52
  18:        0x10da4819d - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h67a1dd621400a14a
  19:        0x10d9e2c2c - tokio::runtime::Runtime::block_on::h924f5d3b891a1855
  20:        0x10db6fd99 - deno::main::h4b91b4803678a6ff
  21:        0x10d96e1fc - std::sys_common::backtrace::__rust_begin_short_backtrace::hf1ae54cd272ba604
  22:        0x10db7b681 - _main
```

### Environment

macOS Catalina 10.15.7

Installed deno via `brew install deno`.

```
$ deno --version
deno 1.6.0 (release, x86_64-apple-darwin)
v8 8.8.278.2
typescript 4.1.2
```

**Comment by satyarohith** [comment]
Reproducible with the latest version (`1.6.3`).

**Comment by caspervonb** [comment]
Still reproducible with master.
Most likely because of how we're doing the borrows here.

Related #7400

**Comment by satyarohith** [comment]
Not sure why I'm not seeing the error on master but here's some additional info if it helps.

```sh
➜  deno git:(master) ✗ ./target/debug/deno run --unstable --allow-write bug.ts
error: Uncaught (in promise) NotFound: No such file or directory (os error 2)
    at processResponse (deno:core/core.js:223:11)
    at Object.jsonOpAsync (deno:core/core.js:240:12)
    at async Object.utime (deno:runtime/js/30_fs.js:337:5)
➜  deno run --unstable --allow-write bug.ts
thread 'main' panicked at 'already borrowed: BorrowMutError', runtime/ops/fs.rs:198:53
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
➜  rustc --version
rustc 1.50.0-nightly (0edce6f4b 2020-12-24)
```
Running on macOS Catalina 10.15.7

**Comment by caspervonb** [comment]
Actually, this is not #7400 because it's the top level state.borrow that's being raced and panics.

**Comment by bartlomieju** [comment]
@piscisaureus please take a look at this issue

**Comment by bartlomieju** [comment]
This problem was fixed in https://github.com/denoland/deno/pull/8955 and will be released in Deno 1.7.0. I'll keep the issue open until then.

**Comment by caspervonb** [comment]
Yeah so I was on v1.47 yesterday when I was looking into this. Bumped rustup and rebuilt today, seems to be all good on master with up to date Rust 👍

**Comment by caspervonb** [comment]
This can be closed now @bartlomieju, double checked on the shipped 1.7.0 and its fine.
